test/test_examples.py

```diff
@@ -442,6 +442,62 @@ def test_cuse(output_checker):
     finally:
         mount_process.terminate()
 
+def test_746(tmpdir, output_checker):
+    """test case for Issue #746
+
+    If RELEASE and UNLINK opcodes are sent back to back, and fuse_fs_release()
+    and fuse_fs_rename() are slow to execute, UNLINK will run while RELEASE is
+    still executing. UNLINK will try to rename the file and, while the rename
+    is happening, the RELEASE will finish executing. As a result, RELEASE will
+    not detect in time that UNLINK has happened, and UNLINK will not detect in
+    time that RELEASE has happened.
+
+
+    NOTE: This is triggered only when nullpath_ok is set.
+
+    If it is NOT SET then get_path_nullok() called by fuse_lib_release() will
+    call get_path_common() and lock the path, and then the fuse_lib_unlink()
+    will wait for the path to be unlocked before executing and thus synchronise
+    with fuse_lib_release().
+
+    If it is SET then get_path_nullok() will just set the path to null and
+    return without locking anything and thus allowing fuse_lib_unlink() to
+    eventually execute unimpeded while fuse_lib_release() is still running.
+    """
+
+    fuse_mountpoint = str(tmpdir)
+
+    fuse_binary_command = base_cmdline + \
+        [ pjoin(basename, 'test', 'issue_746_helper'),
+        "-f", fuse_mountpoint]
+
+    fuse_process = subprocess.Popen(fuse_binary_command,
+                                   stdout=output_checker.fd,
+                                   stderr=output_checker.fd)
+
+    try:
+        wait_for_mount(fuse_process, fuse_mountpoint)
+
+        temp_dir = tempfile.TemporaryDirectory(dir=fuse_mountpoint + "/tmp/")
```

Comment:
Why the need for another temporary directory? The mountpoint is already a temporary directory, so couldn't we do everything there and eliminate the need for the filesystem to implement mkdir() et al?

Reply:
We need a clean empty directory to test for leftovers. Unless I'm mistaken, being a passthrough the temp dir is the "/" of the system, so we need an guaranteed empty subdir with write permissions to create & delete the temp file and then check that the dir is empty. The test uses `/tmp/[random_dir]`.
Is there any pre-existing dir in the CI system that is guaranteed to be empty and have write perms?

Reply:
> We need a clean empty directory to test for leftovers. Unless I'm mistaken, being a passthrough the temp dir is the "/" of the system,

It is for `examples/passthrough.c` but that doesn't need to be the case for the helper. You could e.g. just use the current working directory, and change into `tmpdir` prior to running the helper.